models/headset_encoders/loss.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""A wrapper for universal facial encoder"""
+
+from typing import Any, Dict, List, Tuple, Union
+
+import cv2
+import imageio
+import numpy as np
+import torch
+import torch.nn.functional as F
+from einops import rearrange
+
+from models.headset_encoders.tools import get_color_map
+from models.headset_encoders.ud import FixedFrontalViewGenerator, UDWrapper
+
+
+class UniversalEncoderLoss(torch.nn.Module):
+    def __init__(self, encoder: torch.nn.Module, identities: List[str]) -> None:
+        """
+        A wrapper for running, decoding the evaluating universal headset facial encoders
+
+        Args:
+            encoder: The universal decoder to use.
+            identities: The list of decoder identity names to use.
+        """
+        super().__init__()
+        self._encoder = encoder
+        self._decoder = UDWrapper(ud_exp_name="/uca/leochli/oss/ava256_universal_decoder", identities=identities)
```

Comment:
Should the decoder be passed as an argument?

Reply:
Will fix.